### DISCUSSION TITLE
Calculating a hash for UploadFile

### DISCUSSION THREAD

**Discussion by zevisert** [encode/starlette#1934] [Ideas]
In a project I'm working on, I'd like to have a simple hash of the bytes received for a `multipart/form-data` `UploadFile`, my first instinct knowing that this wasn't something that is directly available from starlette was to subclass the UploadFile class myself to update the hash on seek and write calls, but I don't think that will work since we wan't provide our customized UploadFile subclass back to the `starlette.formparsesrs.MultiPartParser`. 

I can always just read through the file in my route hander myself to calculate the hash, but since the underlaying file is a spooled tempfile, doing to will create a whole extra load of the file from disk if it's been rolled already. Better ideas would be appreciated!

**Comment by adriangb** [comment]
This seems like a reasonable feature request to me. We [do want to support it](https://github.com/encode/starlette/issues/697#issuecomment-549074300). I think maybe #1382 is the first step. Maybe bump that issue?

**Reply by zevisert** [reply]
Thanks, so I am correct in that for now I have to take a second pass over an UploadFile's contents to compute an (eg MD5) hash?

**Reply by adriangb** [reply]
Yes, currently that's what you would need to do.
